Source/items.cpp

```diff
@@ -3678,19 +3679,44 @@ void SpawnTheodore(Point position, bool sendmsg)
 	SpawnRewardItem(IDI_THEODORE, position, sendmsg);
 }
 
+static bool IsStandAtPosition(const Point &position)
+{
+	for (int i = 0; i < ActiveObjectCount; i++) {
+		const Object &object = Objects[ActiveObjects[i]];
+		if (object._otype == OBJ_STAND && object.position == position) {
+			return true;
+		}
+	}
+	return false;
+}
+
 void RespawnItem(Item &item, bool flipFlag)
 {
 	int it = ItemCAnimTbl[item._iCurs];
 	item.setNewAnimation(flipFlag);
-	item._iRequest = false;
-
-	if (IsAnyOf(item._iCurs, ICURS_MAGIC_ROCK, ICURS_TAVERN_SIGN, ICURS_ANVIL_OF_FURY))
-		item._iSelFlag = 1;
-	else if (IsAnyOf(item._iCurs, ICURS_MAP_OF_THE_STARS, ICURS_RUNE_BOMB, ICURS_THEODORE, ICURS_AURIC_AMULET))
-		item._iSelFlag = 2;
-
-	if (item._iCurs == ICURS_MAGIC_ROCK) {
-		PlaySfxLoc(ItemDropSnds[it], item.position);
+	item._iRequest = false; // Item isn't being picked up by a player
+
+	switch (item._iCurs) {
+	case ICURS_TAVERN_SIGN:
+	case ICURS_ANVIL_OF_FURY:
+		item._iSelFlag = SF_FLOOR;
+		break;
+	case ICURS_MAP_OF_THE_STARS:
+	case ICURS_RUNE_BOMB:
+	case ICURS_THEODORE:
+	case ICURS_AURIC_AMULET:
+		item._iSelFlag = SF_ELEVATED;
+		break;
+	case ICURS_MAGIC_ROCK:
+		if (IsStandAtPosition(item.position)) {

```

Comment:
```suggestion
		Object *stand = FindObjectAtPosition(item.position);
		if (stand != nullptr && stand->_oType != OBJ_STAND) {
```

Don't need to iterate through all active objects when we can already look up objects based on a tile position.

Did you change anything to allow the rock to be dropped back onto the stand though? In the video it gets shunted off to the side presumably because CanPut/IsItemBlockingObjectAtPosition considers tiles occupied by objects as blocking items, and there's no exception that would allow the rock to pass the check if the object is the stand.

Reply:
Apply your fix. I didn't change anything to allow the rock to be dropped back onto the stand. I don't believe it should be once picked up to the cursor.